### DISCUSSION TITLE
returning custom responses

### DISCUSSION THREAD

**Discussion by akrasic** [hyperium/hyper#3545] [Q&A] [answered]
Hello all, 

Current setup is running a service that receives the `hyper::body::Incoming` and currently it' set to return `hyper::Response<hyper::body::Incoming>`  which is working as expected. 

What would the the way to return a custom 404 response with some text if, for example, `Host` header doesn't match what I want ? 

For example (mock):

```rust

let host = req.header.s.get("host").unwrap_or(Header::Value::from_status("None"));

if host != "domain.tld" {
  let custom_response = Response::builder().status(404).body("Not found").unwarp();
  Ok(custom_response);
}

```

Thank you for your help and guidance!

**Comment by seanmonstar** [comment, marked as answer]
You may find this guide to be helpful: https://hyper.rs/guides/1/server/echo/

**Reply by akrasic** [reply]
Thank you, that helped a lot.
